CMakeLists.txt

```diff
@@ -41,6 +41,9 @@ include(CMakeDependentOption)
 # Project specific cmake options
 option(COMPILE_WASM "Compile for WASM" OFF)
 cmake_dependent_option(USE_WASM_COMPATIBLE_SOURCE "Use wasm compatible sources" OFF "NOT COMPILE_WASM" ON)
+
+# WASM disables a million libraries, which also includes the unit test-library.
+cmake_dependent_option(COMPILE_UNIT_TESTS "Compile unit tests" OFF "USE_WASM_COMPATIBLE_SOURCE" ON)
```

Comment:
@abhi-agg Opinion on this?  I'm going to let it in for now, but seems we should run tests...

Reply:
The whole idea of creating `USE_WASM_COMPATIBLE_SOURCE` was to run tests. So, yeah tests should run for "minimal" marian. It's worth checking which libraries are missing that stop us from running the test suite (that runs for "full" marian) on "minimal" marian.

@jerinphilip Do you know which libraries are missing for this config? I can try enabling them for "minimal" marian.